src/libraries/System.Collections.Immutable/src/System/Linq/ImmutableArrayExtensions.cs

```diff
@@ -187,28 +187,39 @@ public static bool SequenceEqual<TDerived, TBase>(this ImmutableArray<TBase> imm
         /// <typeparam name="TBase">The type of element contained by the collection.</typeparam>
         public static bool SequenceEqual<TDerived, TBase>(this ImmutableArray<TBase> immutableArray, IEnumerable<TDerived> items, IEqualityComparer<TBase>? comparer = null) where TDerived : TBase
         {
-            Requires.NotNull(items, nameof(items));
+            if (items is ICollection<TBase> itemsCol)
+            {
+                immutableArray.ThrowNullRefIfNotInitialized();
+                return Enumerable.SequenceEqual(immutableArray.array!, itemsCol, comparer);
+            }
 
-            comparer ??= EqualityComparer<TBase>.Default;
+            return Impl(immutableArray, items, comparer);
```

Comment:
If you're making more changes, may as well change the method name to `Enumerate` to be more descriptive.